SAML-OIDC/WebApp_SAML_OIDC/README.md

```diff
@@ -0,0 +1,337 @@
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `WebApp_SAML_OIDC\Web.config` file.
+1. Find the key `ida:ClientId` and replace the existing value with the application ID (clientId) of `Web-App-ASPNET` app copied from the Azure portal.
+1. Find the key `ida:TenantId` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `ida:Domain` and replace the existing value with your Azure AD tenant name.
+
+## Configuring the sample on an AD FS server
+
+### Step 1: Add a Relying Party Trust
+
+1. Log into the server in the on-prem Active Directory domain where AD FS is also installed.
+1. Open the **Server Manager** Dashboard. Under Tools choose **AD FS Management**
+1. Select **Add Relying Party Trust**
+1. Click **Start**
+1. Choose the option **Enter data about relying party manually** and click **Next**
+1. Add a display name, for instance `WebApp_SAML`, and click **Next**
+1. Choose the **AD FS profile** option and click **Next**
+1. Click **Next** on the Configure Certificate step.
+1. Click **Next** on the Configure URL step
+1. Under Relying party trust identifier add your application’s web URL (this sample uses `https://localhost:44347`)
+1. Click **Add** and then click **Next**
+    > For demo purposes, we have an IIS Express development certificate.
+1. Click **Next** on the Configure Multi-factor Authentication if needed
+1. Click **Next** on the Choose Issuance Authorization Rules if needed
+1. Click **Next** on the Ready to Add Trust step if needed.
+1. Close the wizard
+
+### Step 2: Adding a Claim Policy
+
+1. On the AD FS Management window, select **Relying Party Trust**
+1. Select the application name that you have chosen on the previous step and click **Edit Claim Issuance Policy** on the right menu
+1. In the new popup window, click **Add Rule**
+1. Choose the Claim rule template as **Send LDAP Attributes as Claims** and click **Next**
+1. Add a Claim rule name, for instance `Basic Claims`
+1. Select **Active Directory** as the Attribute Store
+1. Configure the LDAP attributes to outgoing claim types. Add the above four LDAP attributes and their corresponding outing claim type:
+    - E-Mail-Addresses -> Email Address
+    - User-Principal-Name -> UPN
+    - Given-Name -> Given Name
+    - Display-Name -> Name
+    - Employee-ID -> EmployeeID
+1. [*Optional*] Feel free to map other claims that are relevant in your scenario, like groups and directory extensions
+1. Click **Finish**
+
+### Step 3: Configure the .NET MVC application
+
+This sample is using the NuGet package **Microsoft.Owin.Security.WsFederation** to configure the authentication with AD FS.
+
+1. Open the **WebApp_SAML_OIDC** application
+1. Open the *Web.config* file and replace the key `ida:ADFSMetadata` value with your AD FS FederationMetadata.xml URI. For instance:
+
+    ```xml
+    <add key="ida:ADFSMetadata" value="https://sts.contoso.com/FederationMetadata/2007-06/FederationMetadata.xml" />
```

Comment:
>ADFSMetadata [](http://example.com/codeflow?start=18&length=12)

a step to retrieve this (if possible) from ADFS management console would be nice

Reply:
I didn't find this address on ADFS server but it would remain same, updated the value of key as below:
`https://[enter_your_ADFS_hostname]/federationmetadata/2007-06/federationmetadata.xml`